provider/pkg/provider/util_test.go

```diff
@@ -1224,6 +1275,93 @@ func TestPruneSlice(t *testing.T) {
 				nil,
 			},
 		},
+		{
+			name:        "map slice with empty non-nil value",
+			description: "a slice of map values that include an empty non-nil value",
+			args: args{
+				source: []any{
+					map[string]any{
+						"a": "a",
+						"b": "b",
+					},
+					map[string]any{},
+				},
+				target: []any{
+					map[string]any{
+						"a": "a",
+						"b": "b",
+					},
+					map[string]any{},
+				},
+			},
+			want: []any{
+				map[string]any{
+					"a": "a",
+					"b": "b",
+				},
+				map[string]any{},
+			},
+		},
+		{
+			name:        "map slice with empty non-nil value in source",
```

Comment:
I think you swapped "source" and "target" in this and the following test name and description

Reply:
Good eye. Fixed.